CONTRIBUTING.md

```diff
@@ -1,4 +1,51 @@
+ - There are three different environments the application has been configured to run in: development, testing, and production.
+ - For testing, we are using [Pytest](https://docs.pytest.org/en/latest/).
+ - It is an expectation that all new functionality, changes in behavior, or bug fixes will be covered under **automated tests**.
+
+There are currently three main areas of the application to be familiar with, and possibly four in the future (dependent on how serialization will be handled).
+
+The three main areas are resources, models, and schemas. This is in line with an MVC web application. The resources directory contains all of the controllers/endpoints. Resources is the terminology that Flask-restful uses and it is the terminology used here. Models are the models, that is where the database tables are defined, and methods that are related to fetching data from the database. The Schemas are for the Marshmallow library, and they define how to validate, serialize, and deserialize the data.
+
+A fourth potential main area is the serializers folder which will be used to build the response.
+
+There is one more file to be familiar with and that is the `app.py` file. This is the file that executes when the application starts. This is also where we are describing the routes for the app.
+
+The rest of this section will describe the three main areas, and how each of those areas should be tested.
+
+Models:
+
+Models define the database schema. They define what tables to create and what columns to use. They can also contain other methods that relate to the business logic of the application. All models in this application inherit from the BaseModel class, which adds created_at and updated_at timestamps for all the tables in the database. It also contains methods that are used to find, create, update, and delete database rows. As of this writing, some models still have an init method. However, for most of the models, the init method is not needed and will be removed. This is because Flask-restful provides an init method that works with keyword arguments, and it is recommended to call super if a custom init method is needed. You can see an example of this in the User Model. Currently, all models except for the lease model have a JSON method that defines how to serialize that object. Models can be found in the `models` directory.
+
+Testing Models:
+
+All models should have unit tests that can be found in the `tests/unit` directory. Each model should have tests that test the inherited methods from the BaseModel to ensure that nothing crazy is inadvertently done to change the behavior of the methods that the BaseModel provides. These are easily implemented using the base_interface_test file. Finally, there should be a test for every public method that is defined in the model. This would be the JSON method or any other method defined in the file that is directly used outside of the Models class.
```

Comment:
```suggestion
All models should have unit tests that can be found in the `tests/unit` directory. Each model should have tests that test the inherited methods from the BaseModel to ensure that nothing crazy is inadvertently done to change the behavior of the methods that the BaseModel provides. These are easily implemented using the `base_interface_test` file. Finally, there should be a test for every public method that is defined in the model. This would be the JSON method or any other method defined in the file that is directly used outside of the Models class.
```